ScheduledTask data now DOM based

talvarez0507 opened the pull request:
This PR makes it so all of the ScheduledTask data is now displayed on the DOM. Additionally the function that retrieves the scheduledTasks now uses the DOM since the information is present there. A local variable storing all of the scheduledTasks was removed because it is no longer necessary. The Sheets API data collection was also updated to reflect the changes in how the data is presented.

talvarez0507 committed:
make scheduledTask data DOM based

Now all of the information for a scheduled task is displayed on the DOM and the collectAllScheduledTasks() function also reads the data from the DOM

talvarez0507 committed:
remove hard-coded algorithm

talvarez0507 committed:
update sheets API to new function changes

talvarez0507 committed:
change to 1 param per line

talvarez0507 committed:
remove extra space after line

jamesnam-google commented on portfolio/src/main/webapp/request.js:
```diff
@@ -64,9 +65,46 @@ function onClickStartScheduling() {
+
+    var scheduledTask = {};
+    const task = new Task(
+        scheduledTaskName, scheduledTaskDescription,
```
Consider placing one param per line. For example, when there are many params (like 7), often we end up looking for the 5th param. It's easier to find that when there is one param per line.

talvarez0507 replied:
Done.

jamesnam-google commented on portfolio/src/main/webapp/request.js:
```diff
@@ -64,9 +65,46 @@ function onClickStartScheduling() {
+    // The cardBody of the scheduledTaskCard
+    const cardBody = scheduledTaskCard.childNodes[0];
+
+    const scheduledTaskName = cardBody.childNodes[0].innerText;
```
I would suggest that you store the data in some HTML element attribute, rather than inner text.

For example, the scheduled task name HTML element can have a data attribute, like,

data-task-name="..."

Then you don't have to do this substring operation. See the pages below for more info.

https://www.w3schools.com/tags/att_data-.asp
https://api.jquery.com/data/

talvarez0507 replied:
Done

talvarez0507 committed:
change to 1 param per line

talvarez0507 committed:
add data attributes to html elements

Instead of doing string operations, I have updated the cards to include data attributes to retrieve the data

rgcase commented on portfolio/src/main/webapp/request.js:
```diff
@@ -52,10 +54,9 @@ function createScheduleRequestFromDom() {
   // Create the request to send to the server using the data we collected from
   // the web form.
-  fetchScheduledTasksFromServlet(scheduleRequest).then((scheduledTaskArray) => {
+  fetchScheduledTasksFromServlet().then((scheduledTaskArray) => {
```
nit: I know it was already like this before, and it's up to you to decide what you prefer but I think you might be able to just pass the `handleScheduledTaskArray` function directly like this:
```
fetchScheduledTasksFromServlet().then(handleScheduledTaskArray);
```

talvarez0507 replied:
Done.

jamesnam-google commented on portfolio/src/main/webapp/request.js:
```diff
@@ -64,9 +65,46 @@ function onClickStartScheduling() {
+        scheduledTaskDescription,
+        scheduledTaskDurationMinutes,
+        scheduledTaskPriority
+        );
```
nit: This end paren can go to the end of the previous line.

ex:
scheduledTaskPriority);

talvarez0507 replied:
Done.

rgcase commented on portfolio/src/main/webapp/request.js:
```diff
@@ -64,9 +65,46 @@ function onClickStartScheduling() {
+        scheduledTaskDescription,
+        scheduledTaskDurationMinutes,
+        scheduledTaskPriority
+        );
```
nit: no newline before `);`.

talvarez0507 replied:
Done.

jamesnam-google commented on portfolio/src/main/webapp/request.js:
```diff
@@ -115,14 +145,34 @@ function addScheduledTaskToDom(scheduledTaskWrapper, resultElement) {
+  const durationText = document.createElement('p');
+  durationText.classList.add('card-text');
+  durationText.innerText = DURATION_INITIAL_TEXT + taskDurationMinutes;
+  durationText.setAttribute('data-task-duration', taskDurationMinutes);
```
Consider renaming the attribute name to data-task-duration-minutes so the time unit is clear.

talvarez0507 replied:
Done.

jamesnam-google commented on portfolio/src/main/webapp/sheetsApi.js:
```diff
@@ -147,16 +147,13 @@ function singleScheduledTaskToSheetsArray(scheduledTask) {
-  const taskDate = new Date();
-  // This is x1000 because the functions takes milliseconds
-  taskDate.setTime(taskTimeSeconds * 1000);
+  const taskDurationMinutes = task.duration;
```
(Optional) I think task.duration should be task.durationMinutes instead. Feel free to add TODO comment if you want to make that change later.

talvarez0507 replied:
I will leave a TODO.

jamesnam-google reviewed (approved):
Approving the change with comments

talvarez0507 committed:
rewrite fetch call

talvarez0507 committed:
remove extra new line

talvarez0507 committed:
'duration' to 'duration-minutes' on data attribute
